BLD: add setuptools; fix license lint

ken-lauer opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

ken-lauer committed:
BLD: add setuptools; fix license lint

ken-lauer commented:
@conda-forge-admin, please rerender

ken-lauer committed:
BLD: no 3.13 support just yet

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.43.0, and conda-forge-pinning 2024.10.23.15.59.32

ken-lauer committed:
BLD: fix hash

ken-lauer committed:
BLD: add <3.13 pin again - this is required, conda-forge-bot

We can't even import `pint` in 3.13 due to bug in pint
https://github.com/hgrecco/pint/issues/2065 - which means we can't build
with it here.

ColwynGulliford reviewed (approved):
Looks ready.
